openeo-geotrellis/src/test/scala/org/openeo/geotrellis/file/Sentinel3PyramidFactoryTest.scala

```diff
@@ -0,0 +1,110 @@
+package org.openeo.geotrellis.file
+
+import geotrellis.proj4.{CRS, LatLng}
+import geotrellis.raster.{CellSize, MultibandTile, Raster, ShortUserDefinedNoDataCellType}
+import geotrellis.raster.io.geotiff.MultibandGeoTiff
+import geotrellis.raster.testkit.RasterMatchers
+import geotrellis.spark._
+import geotrellis.spark.util.SparkUtils
+import geotrellis.vector.Extent
+import org.apache.spark.SparkContext
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api._
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.{Arguments, MethodSource}
+import org.openeo.geotrellis.ProjectedPolygons
+import org.openeo.geotrelliscommon.DataCubeParameters
+import org.openeo.opensearch.backends.OscarsClient
+
+import java.net.URL
+import java.util
+import java.util.stream.{Stream => JStream}
+
+object Sentinel3PyramidFactoryTest {
+  private var sc: SparkContext = _
+
+  @BeforeAll
+  def setupSpark(): Unit =
+    sc = SparkUtils.createLocalSparkContext("local[*]", appName = classOf[Sentinel3PyramidFactoryTest].getName)
+
+  @AfterAll
+  def tearDownSpark(): Unit = sc.stop()
+
+  def bands: JStream[Arguments] = JStream.of(
+    Arguments.of(util.Collections.singletonList("NDVI")),
+    Arguments.of(util.Arrays.asList("B0", "B2", "B3", "MIR", "og", "ag", "saa", "sm", "sza", "vaa", "vza", "wvg", "tg",
+      "NDVI", "TOA_NDVI"))
+  )
+
+  def testIntroducingEmptyBandRetainsCellTypeArgumentsProvider: JStream[Arguments] = JStream.of(
+    Arguments.of(util.Arrays.asList("vaa", "B0", "NDVI"), java.lang.Boolean.FALSE),
+    Arguments.of(util.Arrays.asList("vaa", "B0", "NDVI", "TOA_NDVI"), java.lang.Boolean.TRUE),
+  )
+}
+
+class Sentinel3PyramidFactoryTest extends RasterMatchers {
+
+  @Disabled("only visual inspection")
+  @ParameterizedTest
+  @MethodSource(Array("bands"))
```

Comment:
Just retain the set of bands that includes TOA_NDVI and ditch the `ParameterizedTest`.